.github/workflows/differential-shellcheck.yml

```diff
@@ -0,0 +1,26 @@
+name: Differential ShellCheck
+on:
+  pull_request:
+    branches: [main]
+
+permissions:
+  contents: read
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+
+    permissions:
+      security-events: write
+      pull-requests: write
```

Comment:
why?

Reply:
[Permissions array](https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs) explicitly specify which permissions will have `GITHUB_TOKEN`.

`security-events: write`: allows to upload SARIF report to GitHub for further processing.
`pull-requests: write`: allows to GitHub bot to comment on PR (please see images above).